### PR TITLE
Use RocksDB for JMT Storage in tests

### PR DESCRIPTION
**Pull request by preston-evans98**
This commit revises the `Storage` trait with a configurable `new` function instead of relying on Default. It updates the `Genesis` trait and macro to use this new functionality, and it replaces the stubbed JmtStorage type with a real implementation backed by RocksDB.

### CODE REVIEW

File: sov-modules/sov-modules-impl/integration-tests/tests/demo/mod.rs [hunk 1]
```diff
@@ -40,13 +40,30 @@ trait DispatchCall {
...
 // #[derive(Genesis, DispatchCall, DispatchQuery, Client)]
-#[derive(Genesis)]
+// #[derive(Genesis)]
 struct Runtime<C: Context> {
     election: Election<C>,
     value_adder: ValueSetter<C>,
     //..
 }
 
+impl<C: Context> Genesis for Runtime<C> {
```
Comment (bkolad) [inline sov-modules/sov-modules-impl/integration-tests/tests/demo/mod.rs:50]: We can remove this and enable #[derive(Genesis)].

File: sov-modules/sov-state/src/jmt_storage.rs [hunk 2]
```diff
@@ -34,3 +44,30 @@ impl JmtStorage {
         cache.get_first_reads()
     }
 }
+
+impl Storage for JmtStorage {
+    /// Instead of a config object, we just pass in the
+    /// db instance directly for now.
+    //  TODO: decide whether to use an actual config, or
+    //  rename this type
+    type Config = JmtDb;
```
Comment (bkolad) [inline sov-modules/sov-state/src/jmt_storage.rs:53]: I think the config should propagate via `Spec` trait, but we can do it later.

File: sov-modules/sov-state/src/storage.rs [hunk 3]
```diff
@@ -105,7 +108,13 @@ pub struct GenericStorage<VR: ValueReader> {
     pub(crate) internal_cache: StorageInternalCache,
 }
 
-impl<VR: ValueReader> Storage for GenericStorage<VR> {
+impl<VR: ValueReader + Default> Storage for GenericStorage<VR> {
```
Comment (bkolad) [inline sov-modules/sov-state/src/storage.rs:111]: Let's remove the `GenericStorage` type and implement `ZkStorage` directly, the same way as `JmtStorage`.

**Review by bkolad — APPROVED** [review]

File: sov-modules/sov-modules-impl/integration-tests/tests/demo/mod.rs (hunk 1 above)
Comment (preston-evans98) [reply sov-modules/sov-modules-impl/integration-tests/tests/demo/mod.rs:50]: Done!

**Review by bkolad — APPROVED** [review]

**Pull request merged**